### ISSUE TITLE
Pre-trained weights are not fully compatible with backbone

### ISSUE THREAD

**Issue by rulixiang** [SwinTransformer/Swin-Transformer-Semantic-Segmentation#13]
Hey! Thanks for your brilliant work!

The pre-trained weights seemed to be not fully compatible with the backbone. What's the reason for this difference? Does it affect the training process?

```
Missing key(s) in state_dict: "absolute_pos_embed", "norm0.weight", "norm0.bias", "norm1.weight", "norm1.bias", "norm2.weight", "norm2.bias", "norm3.weight", "norm3.bias". 
Unexpected key(s) in state_dict: "norm.weight", "norm.bias", "layers.0.blocks.1.attn_mask", "layers.1.blocks.1.attn_mask", "layers.2.blocks.1.attn_mask", "layers.2.blocks.3.attn_mask", "layers.2.blocks.5.attn_mask".
```

**Comment by impiga** [comment]
Hi, @rulixiang

#### 1. absolute position embedding

We do not use absolute position embedding by default. You may refer to Table 4 in our paper for ablations.

#### 2. norm layers

For pre-training with classification task, only the output of the last layer is used and we apply a norm layer after it(see [this line](https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer.py#L568)).

While for segmentation, we use outputs from different layers. After each output, we add an extra norm layer(see [this line](https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation/blob/60ebe1a23f9280bd43a1fc0a32b6e78aa8671cef/mmseg/models/backbones/swin_transformer.py#L620)).

This difference results in the missing keys of `norm0.weight`, `norm0.bias`, .... and the unexpected keys of `norm.weight`, `norm.bias`.

#### 3. attention masks

Attention masks is implemented for *cyclic shifted window*. 

Note that input image sizes are fixed in the classification task, so these masks are computed in the initialization and registered in model buffer.

For segmentation, input image sizes may differ in different forward pass. So we compute attention masks on the fly.

**Comment by LiuNull** [comment]
mark

**Comment by lzl2040** [comment]
> 

请问swin transformer作为backbone可以处理不同尺寸的输入吗，是否能和resnet一样呢？我注意到它的参数里面是有img_size这一项的，那在测试过程中对于不同尺寸的图像怎么办？

**Comment by lzl2040** [comment]
> Hi, @rulixiang
> 
> #### 1. absolute position embedding
> We do not use absolute position embedding by default. You may refer to Table 4 in our paper for ablations.
> 
> #### 2. norm layers
> For pre-training with classification task, only the output of the last layer is used and we apply a norm layer after it(see [this line](https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer.py#L568)).
> 
> While for segmentation, we use outputs from different layers. After each output, we add an extra norm layer(see [this line](https://github.com/SwinTransformer/Swin-Transformer-Semantic-Segmentation/blob/60ebe1a23f9280bd43a1fc0a32b6e78aa8671cef/mmseg/models/backbones/swin_transformer.py#L620)).
> 
> This difference results in the missing keys of `norm0.weight`, `norm0.bias`, .... and the unexpected keys of `norm.weight`, `norm.bias`.
> 
> #### 3. attention masks
> Attention masks is implemented for _cyclic shifted window_.
> 
> Note that input image sizes are fixed in the classification task, so these masks are computed in the initialization and registered in model buffer.
> 
> For segmentation, input image sizes may differ in different forward pass. So we compute attention masks on the fly.

请问swin transformer作为backbone可以处理不同尺寸的输入吗，是否能和resnet一样呢？我注意到它的参数里面是有img_size这一项的，那在测试过程中对于不同尺寸的图像怎么办？

**Comment by xuxiansheng2018** [comment]
我的警告如下 ，应该如何设置呢。   此外 该警告会产生影响吗
 ppocr WARNING: backbone.pos_embed not in loaded params dict_keys(['backbone.patch_embed.proj.0.conv.weight', 'backbone.patch_embed.proj.0.conv.bias', 'backbone.patch_embed.proj.0.norm.weight', 'backbone.patch_embed.proj.0.norm.bias', 'backbone.patch_embed.proj.0.norm._mean', 'backbone.patch_embed.proj.0.norm._variance', 'backbone.patch_embed.proj.1.conv.weight', 'backbone.patch_embed.proj.1.conv.bias', 'backbone.patch_embed.proj.1.norm.weight', 'backbone.patch_embed.proj.1.norm.bias', 'backbone.patch_embed.proj.1.norm._mean', 'backbone.patch_embed.proj.1.norm._variance', 'backbone.blocks1.0.norm1.weight', 'backbone.blocks1.0.norm1.bias', 'backbone.blocks1.0.mixer.qkv.weight', 'backbone.blocks1.0.mixer.qkv.bias', 'backbone.blocks1.0.mixer.proj.weight', 'backbone.blocks1.0.mixer.proj.bias', 'backbone.blocks1.0.norm2.weight', 'backbone.blocks1.0.norm2.bias', 'backbone.blocks1.0.mlp.fc1.weight', 'backbone.blocks1.0.mlp.fc1.bias', 'backbone.blocks1.0.mlp.fc2.weight', 'backbone.blocks1.0.mlp.fc2.bias', 'backbone.blocks1.1.norm1.weight', 'backbone.blocks1.1.norm1.bias', 'backbone.blocks1.1.mixer.qkv.weight', 'backbone.blocks1.1.mixer.qkv.bias', 'backbone.blocks1.1.mixer.proj.weight', 'backbone.blocks1.1.mixer.proj.bias', 'backbone.blocks1.1.norm2.weight', 'backbone.blocks1.1.norm2.bias', 'backbone.blocks1.1.mlp.fc1.weight', 'backbone.blocks1.1.mlp.fc1.bias', 'backbone.blocks1.1.mlp.fc2.weight', 'backbone.blocks1.1.mlp.fc2.bias', 'backbone.blocks1.2.norm1.weight', 'backbone.blocks1.2.norm1.bias', 'backbone.blocks1.2.mixer.qkv.weight', 'backbone.blocks1.2.mixer.qkv.bias', 'backbone.blocks1.2.mixer.proj.weight', 'backbone.blocks1.2.mixer.proj.bias', 'backbone.blocks1.2.norm2.weight', 'backbone.blocks1.2.norm2.bias', 'backbone.blocks1.2.mlp.fc1.weight', 'backbone.blocks1.2.mlp.fc1.bias', 'backbone.blocks1.2.mlp.fc2.weight', 'backbone.blocks1.2.mlp.fc2.bias', 'backbone.sub_sample1.conv.weight', 'backbone.sub_sample1.conv.bias', 'backbone.sub_sample1.norm.weight', 'backbone.sub_sample1.norm.bias', 'backbone.blocks2.0.norm1.weight', 'backbone.blocks2.0.norm1.bias', 'backbone.blocks2.0.mixer.qkv.weight', 'backbone.blocks2.0.mixer.qkv.bias', 'backbone.blocks2.0.mixer.proj.weight', 'backbone.blocks2.0.mixer.proj.bias', 'backbone.blocks2.0.norm2.weight', 'backbone.blocks2.0.norm2.bias', 'backbone.blocks2.0.mlp.fc1.weight', 'backbone.blocks2.0.mlp.fc1.bias', 'backbone.blocks2.0.mlp.fc2.weight', 'backbone.blocks2.0.mlp.fc2.bias', 'backbone.blocks2.1.norm1.weight', 'backbone.blocks2.1.norm1.bias', 'backbone.blocks2.1.mixer.qkv.weight', 'backbone.blocks2.1.mixer.qkv.bias', 'backbone.blocks2.1.mixer.proj.weight', 'backbone.blocks2.1.mixer.proj.bias', 'backbone.blocks2.1.norm2.weight', 'backbone.blocks2.1.norm2.bias', 'backbone.blocks2.1.mlp.fc1.weight', 'backbone.blocks2.1.mlp.fc1.bias', 'backbone.blocks2.1.mlp.fc2.weight', 'backbone.blocks2.1.mlp.fc2.bias', 'backbone.blocks2.2.norm1.weight', 'backbone.blocks2.2.norm1.bias', 'backbone.blocks2.2.mixer.qkv.weight', 'backbone.blocks2.2.mixer.qkv.bias', 'backbone.blocks2.2.mixer.proj.weight', 'backbone.blocks2.2.mixer.proj.bias', 'backbone.blocks2.2.norm2.weight', 'backbone.blocks2.2.norm2.bias', 'backbone.blocks2.2.mlp.fc1.weight', 'backbone.blocks2.2.mlp.fc1.bias', 'backbone.blocks2.2.mlp.fc2.weight', 'backbone.blocks2.2.mlp.fc2.bias', 'backbone.blocks2.3.norm1.weight', 'backbone.blocks2.3.norm1.bias', 'backbone.blocks2.3.mixer.qkv.weight', 'backbone.blocks2.3.mixer.qkv.bias', 'backbone.blocks2.3.mixer.proj.weight', 'backbone.blocks2.3.mixer.proj.bias', 'backbone.blocks2.3.norm2.weight', 'backbone.blocks2.3.norm2.bias', 'backbone.blocks2.3.mlp.fc1.weight', 'backbone.blocks2.3.mlp.fc1.bias', 'backbone.blocks2.3.mlp.fc2.weight', 'backbone.blocks2.3.mlp.fc2.bias', 'backbone.blocks2.4.norm1.weight', 'backbone.blocks2.4.norm1.bias', 'backbone.blocks2.4.mixer.qkv.weight', 'backbone.blocks2.4.mixer.qkv.bias', 'backbone.blocks2.4.mixer.proj.weight', 'backbone.blocks2.4.mixer.proj.bias', 'backbone.blocks2.4.norm2.weight', 'backbone.blocks2.4.norm2.bias', 'backbone.blocks2.4.mlp.fc1.weight', 'backbone.blocks2.4.mlp.fc1.bias', 'backbone.blocks2.4.mlp.fc2.weight', 'backbone.blocks2.4.mlp.fc2.bias', 'backbone.blocks2.5.norm1.weight', 'backbone.blocks2.5.norm1.bias', 'backbone.blocks2.5.mixer.qkv.weight', 'backbone.blocks2.5.mixer.qkv.bias', 'backbone.blocks2.5.mixer.proj.weight', 'backbone.blocks2.5.mixer.proj.bias', 'backbone.blocks2.5.norm2.weight', 'backbone.blocks2.5.norm2.bias', 'backbone.blocks2.5.mlp.fc1.weight', 'backbone.blocks2.5.mlp.fc1.bias', 'backbone.blocks2.5.mlp.fc2.weight', 'backbone.blocks2.5.mlp.fc2.bias', 'backbone.sub_sample2.conv.weight', 'backbone.sub_sample2.conv.bias', 'backbone.sub_sample2.norm.weight', 'backbone.sub_sample2.norm.bias', 'backbone.blocks3.0.norm1.weight', 'backbone.blocks3.0.norm1.bias', 'backbone.blocks3.0.mixer.qkv.weight', 'backbone.blocks3.0.mixer.qkv.bias', 'backbone.blocks3.0.mixer.proj.weight', 'backbone.blocks3.0.mixer.proj.bias', 'backbone.blocks3.0.norm2.weight', 'backbone.blocks3.0.norm2.bias', 'backbone.blocks3.0.mlp.fc1.weight', 'backbone.blocks3.0.mlp.fc1.bias', 'backbone.blocks3.0.mlp.fc2.weight', 'backbone.blocks3.0.mlp.fc2.bias', 'backbone.blocks3.1.norm1.weight', 'backbone.blocks3.1.norm1.bias', 'backbone.blocks3.1.mixer.qkv.weight', 'backbone.blocks3.1.mixer.qkv.bias', 'backbone.blocks3.1.mixer.proj.weight', 'backbone.blocks3.1.mixer.proj.bias', 'backbone.blocks3.1.norm2.weight', 'backbone.blocks3.1.norm2.bias', 'backbone.blocks3.1.mlp.fc1.weight', 'backbone.blocks3.1.mlp.fc1.bias', 'backbone.blocks3.1.mlp.fc2.weight', 'backbone.blocks3.1.mlp.fc2.bias', 'backbone.blocks3.2.norm1.weight', 'backbone.blocks3.2.norm1.bias', 'backbone.blocks3.2.mixer.qkv.weight', 'backbone.blocks3.2.mixer.qkv.bias', 'backbone.blocks3.2.mixer.proj.weight', 'backbone.blocks3.2.mixer.proj.bias', 'backbone.blocks3.2.norm2.weight', 'backbone.blocks3.2.norm2.bias', 'backbone.blocks3.2.mlp.fc1.weight', 'backbone.blocks3.2.mlp.fc1.bias', 'backbone.blocks3.2.mlp.fc2.weight', 'backbone.blocks3.2.mlp.fc2.bias', 'backbone.last_conv.weight', 'backbone.norm.weight', 'backbone.norm.bias', 'head.fc.weight', 'head.fc.bias']) !
